py/specter/psf/spotgrid.py

```diff
@@ -57,16 +58,37 @@ def _xypix(self, ispec, wavelength):
         Return xslice, yslice, pix for PSF at spectrum ispec, wavelength
         """
         return self._xypix_interp(ispec, wavelength)
+        
+    @jit(nopython=True,cache=True)
+    def new_pixshift(dx,dy,w00,w10,w01,w11,pix_spot_values,rebin):
+        """
+        Return resampled_pix_spot_values 
+        """
+        ny_spot, nx_spot = pix_spot_values.shape
+        #preallocate 
+        resampled_pix_spot_values=np.zeros((ny_spot+rebin,nx_spot+rebin))
+        for i in range(0,ny_spot):
+            for j in range(0,nx_spot):  
+                resampled_pix_spot_values[dy+i,dx+j]       += w00*pix_spot_values[i,j] 
+                resampled_pix_spot_values[dy+1+i,dx+j]     += w10*pix_spot_values[i,j]
+                resampled_pix_spot_values[dy+i,dx+1+j]     += w01*pix_spot_values[i,j]
+                resampled_pix_spot_values[dy+1+i,dx+1+j]   += w11*pix_spot_values[i,j] 
+                    
+        return resampled_pix_spot_values
```

Comment:
in `_xypix_interp` we originally did 2 steps: resampling to a different high resolution grid and then rebinning to a coarser grid:
```python
# resampled spot grid
resampled_pix_spot_values=SpotGridPSF.new_pixshift(...)       
            
# rebinning
ccd_pix_spot_values=resampled_pix_spot_values.reshape(ny_spot+rebin,nx_ccd,rebin).sum(2).reshape(ny_ccd,rebin,nx_ccd).sum(1)
```
I'm guessing that you could get another speed win by going ahead and doing the rebinning while resampling within this function, thus avoiding the need to create the big `resampled_pix_spot_values` only to immediately rebin it.  Double check and test this, but I think that would be:
```python
nx_binned = (nx_spot+rebin) // rebin
ny_binned = (ny_spot+rebin) // rebin
rebinned_pix_spot_values=np.zeros((ny_binned, nx_binned))
for i in range(ny_spot):
    for j in range(nx_spot):  
        rebinned_pix_spot_values[(dy+i)//rebin,  (dx+j)//rebin]   += w00*pix_spot_values[i,j] 
        rebinned_pix_spot_values[(dy+1+i)//rebin,(dx+j)//rebin]   += w10*pix_spot_values[i,j]
        rebinned_pix_spot_values[(dy+i)//rebin,  (dx+1+j)//rebin] += w01*pix_spot_values[i,j]
        rebinned_pix_spot_values[(dy+1+i)//rebin,(dx+1+j)//rebin] += w11*pix_spot_values[i,j] 
```
That would be terribly inefficient normal python code with the nested loops, but that works well for numba and could work well to resample and rebin simultaneously.  Please try that.

Reply:
Thank you for the suggestion. I tested this segment and it provides ALMOST the same result as the original method. Based on some floating point division (I think), there are differences between 1E-13 to 1E-16 in the resulting matrices from the two methods. Is difference acceptable for you?

Reply:
Yes, that's fine.  When you're finished making changes, add a note to this PR that it is ready for re-review, and comment on whether this particular refactor gave additional speedup or in the end didn't matter.  Thanks.

Reply:
I did test this idea, but it runs much more slowly than the original version. Therefore these changes were not included.